src/App.tsx

```diff
@@ -1,24 +1,203 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './Components/ToDoList';
+import { Notification } from './Components/errorNotification';
+import { client } from './utils/fetchClient';
+import { Error } from './types/Error';
 
-const USER_ID = 0;
+import { FilterType, Todo } from './types/Todo';
+
+import { getTodos, deleteTodo } from './api/todos';
+
+import { Footer } from './Components/Footer';
+
+const USER_ID = 6340;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterType>(FilterType.ALL);
+  const [error, setError] = useState<Error>(Error.NONE);
+  const [inputQuery, setInputQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    loadTodosData();
+  }, []);
+
+  const loadTodosData = async () => {
+    try {
+      setError(Error.NONE);
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setError(Error.LOAD);
+    }
+  };
+
+  const addNewTodo = async () => {
+    if (!inputQuery.trim()) {
+      setError(Error.NONE);
+
+      return;
+    }
+
+    try {
+      setError(Error.NONE);
+      setIsLoading(true);
+
+      const newTodo = {
+        userId: USER_ID,
+        title: inputQuery.trim(),
+        completed: false,
+      };
+
+      const response = await client.post<Todo>('/todos', newTodo);
+
+      setTodos([...todos, response]);
+      setInputQuery('');
+    } catch {
+      setError(Error.ADD);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const deleteTodoItem = async (todoId: number) => {
+    try {
+      setError(Error.NONE);
+      await deleteTodo(todoId);
+
+      setTodos(todos.filter(todo => todo.id !== todoId));
+    } catch {
+      setError(Error.DELETE);
+    }
+  };
+
+  const handleFormSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    addNewTodo();
+  };
+
+  const handleTodoDelete = (todoId: number) => {
+    deleteTodoItem(todoId);
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputQuery(e.target.value);
+  };
+
+  const handleToggleAll = () => {
+    const areAllCompleted = todos.every(todo => todo.completed);
+
+    const updatedTodos = todos.map(todo => ({
+      ...todo,
+      completed: !areAllCompleted,
+    }));
+
+    setTodos(updatedTodos);
+  };
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filterBy) {
+        case FilterType.ACTIVE:
+          return !todo.completed;
+
+        case FilterType.COMPLETED:
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    });
```

Comment:
No need to filter in case FilterType is All
```suggestion
  const visibleTodos = useMemo(() => {
      switch (filterBy) {
        case FilterType.ACTIVE:
          return todos.filter((todo) =>!todo.completed);

        case FilterType.COMPLETED:
          return todos.filter((todo) => todo.completed);

        default:
          return todos;
      }
```